commands/addPermissions.ts

```diff
@@ -0,0 +1,59 @@
+import { ApplicationCommand, ApplicationCommandPermissionData, Collection, CommandInteraction } from 'discord.js';
+import { SlashCommandBuilder, roleMention } from '@discordjs/builders';
+import { ICommand } from '../types/command';
+import { addCommandPermission } from '../common/discordutil';
+
+export const addPermissions: ICommand = {
+    config: new SlashCommandBuilder()
+        .setName('addpermissions')
+        .setDescription('Add permissions to command')
+        .setDefaultPermission(false)
+        .addRoleOption((option) =>
+            option.setName('role').setDescription('The role to add permission').setRequired(true)
+        )
+        .addStringOption((option) =>
+            option.setName('command').setDescription('Name of the command to update permissions').setRequired(true)
+        ),
+    execute: async (interaction: CommandInteraction) => {
+        await interaction.deferReply();
+        const { guild, options } = interaction;
+        const roleID = options.getRole('role').id;
+        const commandName = options.getString('command');
+
+        let commands = new Collection<string, ApplicationCommand>();
```

Comment:
dumb question - why do we need to initialize it as this Collection?

Reply:
because in the try below, guild.commands.fetch() returns a Collection and we are storing as `commands`. 😉